### DISCUSSION TITLE
¿How set different background on pdfmake-wrapper's page?

### DISCUSSION THREAD

**Discussion by AbisurDiazR** [Lugriz/pdfmake-wrapper#58] [Q&A] [answered]
I am working with pdfmake-wrapper on Angular 13, but I don't understand how set a different background per page like this sample:

![image](https://user-images.githubusercontent.com/56562635/149870950-7e300138-a10c-4194-b0b4-d4ea14504ccf.png)

That is my method to generate my own pdf:

```
public async generateFormat() {
    let imageOne!: any;
    let imageTwo!: any;
    let ancho!: any;
    let alto!: any;
    if (this.licencia.format === 'Municipal') {
      imageOne = this.formatoObject.licFrenteMunicipio;
      imageTwo = this.formatoObject.licVueltaMunicipio;
      ancho = 85;
      alto = 54;
    } else {
      imageOne = this.formatoObject.licFrenteEstado;
      imageTwo = this.formatoObject.licVueltaEstado;
      ancho = 54;
      alto = 85;
    }
    const pdf = new PdfMakeWrapper();
    PdfMakeWrapper.setFonts(pdfFonts);
    pdf.add(await new Img(imageOne).width(ancho).height(alto).build().then(async img => {
      pdf.add(await new Img(this.licencia.photo).width(25).height(25).absolutePosition(25,25).build());
      pdf.background(img);
      pdf.pageSize({
        width: ancho,
        height: alto
      })
    }));
    pdf.add('\n');
    pdf.add(await new Img(imageTwo).width(ancho).height(alto).build().then(async img => {
      pdf.add(await new Img(this.licencia.firm).width(25).height(25).absolutePosition(25,25).build());
      pdf.background(img);
      pdf.pageSize({
        width: ancho,
        height: alto
      })
    }));
    pdf.create().open();
  }
```
And this is my result when I try it:

![image](https://user-images.githubusercontent.com/56562635/149870982-110a89be-6bef-4850-96d3-698bba568685.png)

**Comment by Lugriz** [comment, marked as answer]
Thanks for using pdfmake-wrapper! You can try to use [background method](https://github.com/Lugriz/pdfmake-wrapper#backgroundbackground-any---void)  like this:  [https://stackblitz.com/edit/typescript-udur48](https://stackblitz.com/edit/typescript-udur48)

**Reply by AbisurDiazR** [reply]
Thank you very much, this solution helped me a lot, but my current problem is that I don't know how to correctly arrange the text elements in the pages of the PDF document that I am generating, since the page dimensions that I am asked to create this PDF are quite small and they end up overflowing the elements.
